### PR TITLE
Docker Compose: Use Generic Worker Multiuser Engine

### PR DESCRIPTION
**Pull request by petemoore**
Note, this depends on #7247.

### CODE REVIEW

**Comment by petemoore** [comment]
Putting in draft while I test locally. For example, I see that `sudo` is needed in docker image...

File: generic-worker.Dockerfile [hunk 1]
```diff
@@ -53,7 +53,7 @@ else \
     echo \{\"version\": \"69.0.1\", \"commit\": \"local\", \"source\": \"https://github.com/taskcluster/taskcluster\", \"build\": \"NONE\"\} > /version.json; \
 fi
 
-VOLUME /etc/generic-worker/config.json
```
Comment (petemoore) [inline generic-worker.Dockerfile:56]: I'm not sure how this worked before, aiui VOLUME should be a directory?

Comment (lotas) [reply generic-worker.Dockerfile:56]: good catch!

File: workers/generic-worker/multiuser_posix.go [hunk 2]
```diff
@@ -36,12 +36,12 @@ func platformFeatures() []Feature {
 
 func deleteDir(path string) error {
 	log.Print("Removing directory '" + path + "'...")
-	err := host.Run("/usr/bin/sudo", "/bin/chmod", "-R", "u+w", path)
+	err := host.Run("/bin/chmod", "-R", "u+w", path)
```
Comment (lotas) [inline workers/generic-worker/multiuser_posix.go:39]: will it cause any issues for the non docker-compose workers? Or will this always be run with the correct permissions?

**Review by lotas — APPROVED** [review]
nice!

**Pull request merged**